src/geoapis/vector.py

```diff
@@ -78,25 +78,32 @@ def get_features_inside_catchment(self, layer: int, geometry_type: str):
         catchment_bounds = self.catchment_polygon.geometry.bounds
         feature_collection = self.query_vector_wfs(catchment_bounds, layer, geometry_type)
 
-        # Cycle through each feature getting name and coordinates
-        features = []
+        # Cycle through each feature checking in bounds and getting geometry and properties
+        features = {'geometry': []}
         for feature in feature_collection['features']:
 
             shapely_geometry = shapely.geometry.shape(feature['geometry'])
 
             # check intersection of tile and catchment in LINZ CRS
             if self.catchment_polygon.intersects(shapely_geometry).any():
 
-                # convert any one Polygon MultiPolygons to a straight Polygon
+                # Create 'properties' column headings from first in-bounds vector
+                if len(features['geometry']) == 0:
+                    for key in feature['properties'].keys():
+                        features[key] = []
+
+                # Convert any one Polygon MultiPolygons to a straight Polygon then add to the geometries
```

Comment:
The comment is a bit unclear I think.

Reply:
Thanks @Pooja3894. I've updated it - let me know what you think

Reply:
Yup looks good to me. :)